internal/charmstore/stats.go

```diff
@@ -492,38 +492,52 @@ var LegacyDownloadCountsEnabled = true
 
 // ArchiveDownloadCounts calculates the aggregated download counts for
 // a charm or bundle.
-func (s *Store) ArchiveDownloadCounts(id *router.ResolvedURL) (thisRevision, allRevisions AggregatedCounts, err error) {
-	// TODO (frankban): remove this condition when removing the legacy counts
-	// logic.
-	if LegacyDownloadCountsEnabled {
-		return s.legacyDownloadCounts(id)
-	}
-
+func (s *Store) ArchiveDownloadCounts(id *charm.Reference) (thisRevision, allRevisions AggregatedCounts, err error) {
 	// Retrieve the aggregated stats.
-	thisRevision, err = s.aggregateStats(EntityStatsKey(&id.URL, params.StatsArchiveDownload), false)
+	thisRevision, err = s.aggregateStats(EntityStatsKey(id, params.StatsArchiveDownload), false)
 	if err != nil {
 		err = errgo.Notef(err, "cannot get aggregated count for the specific revision")
 		return
 	}
-	noRevisionId := id.URL
+	noRevisionId := *id
 	noRevisionId.Revision = -1
 	allRevisions, err = s.aggregateStats(EntityStatsKey(&noRevisionId, params.StatsArchiveDownload), true)
 	if err != nil {
 		err = errgo.Notef(err, "cannot get aggregated count for all revisions")
 		return
 	}
+	// TODO (frankban): remove this condition when removing the legacy counts
+	// logic.
+	if LegacyDownloadCountsEnabled {
+		legacyRevision, legacyAll, err := s.legacyDownloadCounts(id)
+		if err != nil {
+			return AggregatedCounts{}, AggregatedCounts{}, err
+		}
+		thisRevision.LastDay += legacyRevision.LastDay
+		thisRevision.LastWeek += legacyRevision.LastWeek
+		thisRevision.LastMonth += legacyRevision.LastMonth
+		thisRevision.Total += legacyRevision.Total
+		allRevisions.LastDay += legacyAll.LastDay
+		allRevisions.LastWeek += legacyAll.LastWeek
+		allRevisions.LastMonth += legacyAll.LastMonth
+		allRevisions.Total += legacyAll.Total
```

Comment:
This code reminded me that we should retrieve more than just totals from legacy CS.